app/CMakeLists.txt

```diff
@@ -28,6 +28,21 @@ if(CELERITAS_BUILD_TESTS)
     set(CELER_PROCESSORS)
   endif()
   set(CELER_APP_DATA_DIR "${CMAKE_CURRENT_SOURCE_DIR}/data")
+
+  # Disable device tests if no CUDA/HIP or if environment is set
+  if(CELERITAS_USE_CUDA OR CELERITAS_USE_HIP)
+    if("$ENV{CELER_DISABLE_DEVICE}")
+      message(WARNING
+        "Disabling GPU testing because CELER_DISABLE_DEVICE is set"
+      )
```

Comment:
This warns every time because the value is updated at configure time and could change from one configure to the next.